client/components/data/query-users/index.jsx

```diff
@@ -37,12 +39,20 @@ class QueryUsers extends Component {
 	}
 
 	request() {
-		this.props.requestUsers( this.props.siteId, this.props.userIds );
+		const { siteId, userIds, currentUserId } = this.props;
+		// Only request users if a list of user Ids is provided. Otherwise, all users would be requested incl. the current user.
+		if ( userIds.length ) {
+			// Ensure that we only request the authors whose information we're lacking, not the current user.
+			const requestIds = userIds.filter( id => id !== currentUserId );
```

Comment:
Minor suggestion: I'd move the filter outside the conditional & check `requestIds.length` above instead to prevent calling `requestUsers` with an empty payload.